elrond.md

```diff
@@ -1082,25 +1148,48 @@ TODO: Implement [reserved keys and read-only runtimes](https://github.com/Elrond
          </instrs>
          <locals> 0 |-> <i32> IDX 1 |-> <i32> OFFSET 2 |-> <i32> LENGTH </locals>
 
+ // extern void      bigIntSetInt64(void* context, int32_t destinationHandle, long long value);
+    rule <instrs> hostCall ( "env" , "bigIntSetInt64" , [ i32  i64  .ValTypes ] -> [ .ValTypes ] )
+               => #setBigIntValue(DEST_IDX, VALUE)
+                  ...
+         </instrs>
+         <locals> 0 |-> <i32> DEST_IDX 1 |-> <i64> VALUE </locals>
+
     // extern void bigIntAdd(void* context, int32_t destination, int32_t op1, int32_t op2);
     rule <instrs> hostCall("env", "bigIntAdd", [ i32 i32 i32 .ValTypes ] -> [ .ValTypes ]) => . ... </instrs>
          <locals> 0 |-> <i32> DST  1 |-> <i32> OP1_IDX  2 |-> <i32> OP2_IDX </locals>
          <bigIntHeap> HEAP => HEAP [DST <- {HEAP[OP1_IDX]}:>Int +Int {HEAP[OP2_IDX]}:>Int] </bigIntHeap>
+      requires OP1_IDX in_keys(HEAP)
+       andBool isInt(HEAP[OP1_IDX])
+       andBool OP2_IDX in_keys(HEAP)
+       andBool isInt(HEAP[OP2_IDX])
 
     // extern void bigIntSub(void* context, int32_t destination, int32_t op1, int32_t op2);
     rule <instrs> hostCall("env", "bigIntSub", [ i32 i32 i32 .ValTypes ] -> [ .ValTypes ]) => . ... </instrs>
          <locals> 0 |-> <i32> DST  1 |-> <i32> OP1_IDX  2 |-> <i32> OP2_IDX </locals>
          <bigIntHeap> HEAP => HEAP [DST <- {HEAP[OP1_IDX]}:>Int -Int {HEAP[OP2_IDX]}:>Int] </bigIntHeap>
+      requires OP1_IDX in_keys(HEAP)
+       andBool isInt(HEAP[OP1_IDX])
```

Comment:
I'm not saying that you should do it in this PR, but it is possible to have typed maps, lists and sets. To do that, you would have to
1. if you use primitive types (Int, Bytes, Bool), wrap them in a normal sort
2. copy the MAP (or LIST or SET) module from `domains.md` and change KItem to whatever types you need
3. other small changes needed to prevent conflicts between symbols.

You can find an example in the `kmxwasm/k-src/map-intw-to-bytesw.k` file in the PR I sent for review (https://github.com/runtimeverification/elrond-wasm/pull/2/files#diff-80a3da10dc7bd2f407036b95a7138303e69894e9701e8ee3781284b0daffa9f6R1)

Reply:
FWIW, I have a script that generates the modules for typed maps, lists and sets. I developed it in my personal time, so I didn't try to make it look nice, but, if there is interest in it, I could try to share it and/or put it somewhere on github.